### PR TITLE
Refactor all table queries to use FilterSet

### PR DESCRIPTION
**Pull request by MrCreosote**
Cleans up very similar code in several places

### CODE REVIEW

**Comment by MrCreosote** [comment]
image
Aw yiss

File: src/service/data_products/samples.py [hunk 1]
```diff
@@ -203,71 +204,49 @@ async def get_samples(
...
+            skip=res.skip,
+            limit=res.limit,
+            count=res.count,
+            match_state=dp_match.state if dp_match else None,
+            selection_state=dp_sel.state if dp_sel else None,
+            fields=res.fields,
+            table=res.table,
+            data=res.data,
+        )
+    else:
```
Comment (Xiangs18) [inline src/service/data_products/samples.py:243]: I would personally avoid `else` here.

Comment (MrCreosote) [reply src/service/data_products/samples.py:243]: Here we have a difference of style - either way is technically correct, but to me having both returns in the if/else block

* makes it easier to read
* makes it easier to add more elif blocks in the future

So I prefer it as is

**Review by Xiangs18 — APPROVED** [review]
:100:

**Pull request merged**